activerecord/lib/active_record/errors.rb

```diff
@@ -59,10 +59,27 @@ class AdapterNotFound < ActiveRecordError
   class AdapterError < ActiveRecordError
     def initialize(message = nil, connection_pool: nil)
       @connection_pool = connection_pool
+
+      if message && connection_pool && multiple_databases?
+        message = "#{message}\n#{connection_info(connection_pool)}"
+      end
+
       super(message)
     end
 
     attr_reader :connection_pool
+
+    private
+      def multiple_databases?
```

Comment:
It's unlikely noticeable and we only execute it when an exception happens so performance is not as critical as for success responses but perhaps for the future - I assume the return value of this method won't change in runtime so we could either initialize it during boot or perhaps better memoize it to avoid hurting boot time